pkg/proxy/iptables/proxier.go

```diff
@@ -984,10 +984,21 @@ func (proxier *Proxier) syncProxyRules() {
 					// If the packet was able to reach the end of firewall chain, then it did not get DNATed.
 					// It means the packet cannot go thru the firewall, then mark it for DROP
 					writeLine(proxier.natRules, append(args, "-j", string(KubeMarkDropChain))...)
+				} else {
+					// Add a reject rule so the behavior is the same for clients going via the load balancer
+					// (immediate reject instead of connection timeout). This also avoids a stale entry in the
+					// contrack table.
+					writeLine(proxier.filterRules,
```

Comment:
I don't think it would work there. This code path is used for the externalIPs feature in services. This is not used for external load balancers (at least we don't see it being used with load balancers on Azure, only with the externalIPs feature). I admit to still have a somehow limited knowledge of the code but it appears to me that the rule is exactly where it should be.

Reply:
Yes that is the correct path. The problem happens with Services of type `LoadBalancer` that has `ExternalIPs` and no endpoints. For Azure (or anything that uses dsr) no traffic will be delivered to node if rule is not in place (and has an `ExternalIP`